nano/node/bootstrap_ascending/service.cpp

```diff
@@ -39,13 +39,9 @@ nano::bootstrap_ascending::service::service (nano::node_config & config_a, nano:
 			auto transaction = ledger.tx_begin_read ();
 			for (auto const & [result, context] : batch)
 			{
-				// Do not try to unnecessarily bootstrap live traffic chains
-				if (context.source == nano::block_source::bootstrap)
-				{
-					release_assert (context.block != nullptr);
-					inspect (transaction, result, *context.block);
-					should_notify = true;
-				}
+				release_assert (context.block != nullptr);
+				inspect (transaction, result, *context.block, context.source);
+				should_notify = true;
```

Comment:
It seems like the only case where should_notify would be false is if the batch is empty. If this isn't possible it seems like this should be a debug_assert.
If it is possible, perhaps we should replace "bool should_notify = false;" with "if (batch.empty) return;" and delete should_notify entirely.

Reply:
It's a leftover from previous code, should_notify is indeed not needed now.